### DISCUSSION TITLE
Any idea to default/enforce JSON responses?

### DISCUSSION THREAD

**Discussion by MaximeBernard** [honojs/hono#2935] [Q&A] [answered]
Hi, I'm coming from koa, played with fastify and they both return JSON responses by default.

I was wondering if I could do something similar in Hono?

My ideal would have been a 
```ts
const app = new Hono({ jsonOnly: true });
```

But maybe I can do this with a middleware?

Out of curiosity, why do you need to `c.json` everywhere?

**Comment by EdamAme-x** [comment]
hi @MaximeBernard 
Yes, you can use middleware.
Middleware can be placed at the tail end, and you can create middleware that checks if the response is in json format and throws an error if it is not.

**Reply by MaximeBernard** [reply]
What do you think about
```ts
import { createMiddleware } from 'hono/factory'

export const jsonMiddleware = () => {
  return createMiddleware(async (c, next) => {
    return c.json(await next());
  })
}
```
?

That way, I had in mind, I could do
```ts
app.get('/posts', () => [{id: 1, title: "Hello"}, {id: 2, title: "World"]);
```

**Reply by EdamAme-x** [reply]
sorry, I understand just now.

**Reply by EdamAme-x** [reply, marked as answer]
hmm...
I think that is possible.
But, we have some breaking problems.
That way of writing is similar to elysia.js.
In Hono, that kind of return is impossible due to types.
Perhaps you can do it by playing with the response in middleware, but I don't recommend it because of the types issue.

**Reply by MaximeBernard** [reply]
I see, thank you for your time 👍
